prediction_prophet/functions/research.py

```diff
@@ -29,7 +29,7 @@ def research(
     logger: t.Union[logging.Logger, "Logger"] = logging.getLogger()
 ) -> str:
     logger.info("Started subqueries generation")
-    queries = generate_subqueries(query=goal, limit=initial_subqueries_limit, model=model, api_key=openai_api_key)
+    queries = generate_subqueries(query=goal, limit=initial_subqueries_limit, model=model, api_key=openai_api_key) if initial_subqueries_limit > 0 else [goal]
```

Comment:
imo makes more sense to move this into the function definition - i.e. at the start:
```
if limit == 0:
  return [query]
```
so don't need to have this logic wherever it's used

Reply:
Sure, no preference on my side